indexing-hadoop/src/main/java/io/druid/indexer/InputRowSerde.java

```diff
@@ -0,0 +1,226 @@
+/*
+* Licensed to Metamarkets Group Inc. (Metamarkets) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. Metamarkets licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.indexer;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Maps;
+import com.google.common.io.ByteArrayDataOutput;
+import com.google.common.io.ByteStreams;
+import com.metamx.common.IAE;
+import com.metamx.common.ISE;
+import com.metamx.common.logger.Logger;
+import io.druid.data.input.InputRow;
+import io.druid.data.input.MapBasedInputRow;
+import io.druid.query.aggregation.Aggregator;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.segment.incremental.IncrementalIndex;
+import io.druid.segment.serde.ComplexMetricSerde;
+import io.druid.segment.serde.ComplexMetrics;
+import org.apache.hadoop.io.ArrayWritable;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.FloatWritable;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.MapWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+
+import java.io.DataInput;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ */
+public class InputRowSerde
+{
+  private static final Logger log = new Logger(InputRowSerde.class);
+
+  private static final Text[] EMPTY_TEXT_ARRAY = new Text[0];
+
+  public static final byte[] toBytes(final InputRow row, AggregatorFactory[] aggs)
+  {
+    try {
+      ByteArrayDataOutput out = ByteStreams.newDataOutput();
+
+      List<String> dimList = row.getDimensions();
+
+      Text[] dims = EMPTY_TEXT_ARRAY;
+      if(dimList != null) {
+        dims = new Text[dimList.size()];
+        for (int i = 0; i < dims.length; i++) {
+          dims[i] = new Text(dimList.get(i));
+        }
+      }
+      StringArrayWritable sw = new StringArrayWritable(dims);
+      sw.write(out);
```

Comment:
I would go with if (!dimlist.isEmpty()). So one can skip the whole array init/for block altogether, if inputrow has no dimensions to deliver.

Is there a reason, why an empty array is written nonetheless, even if dimensions were empty?   Or can that be skipped as well?